src/ripple/ledger/impl/RawStateTable.cpp

```diff
@@ -173,13 +173,18 @@ RawStateTable::apply(RawView& to) const
                 to.rawReplace(item.sle);
                 break;
         }
+
+        keysTouched_.emplace(elem.first);
     }
 }
 
 bool
 RawStateTable::exists(ReadView const& base, Keylet const& k) const
```

Comment:
Vinnie added everything you need already ? :)

Reply:
do you know something I don't lol

Reply:
Back in the day there was direct access to the SHAMap from transactors, iirc